torch/_dynamo/optimizations/training.py

```diff
@@ -74,7 +74,9 @@ def _wrapped_bw_compiler(*args, **kwargs):
 
         try:
             # NB: NOT cloned!
-            with enable_aot_logging():
+            with enable_aot_logging(), torch._dispatch.python.enable_crossref_functionalize(
+                crossref_functionalize
```

Comment:
Would it be more logical to push this into AOT Autograd itself?

Reply:
That's fair. Actually, what do you think of unconditionally running it in aot autograd? These extra checks probably won't be our bottleneck for compile time.

Reply:
I'm mostly worried about the robustness of crossref functionalize, not sure I got the logic entirely right. If we can prove it out I'm amenable.